### PR TITLE
[QML/Tutorials] Classification

### PR DESCRIPTION
**Pull request by debrevitatevitae**
Showcase of
1. Qadence for classification
2. Hybrid QNNs (QNN, linear layer and softmax layer)

Can be an alternative ML pipeline to show besides the existing tutorial on learning differential equations.

### CODE REVIEW

**Comment by debrevitatevitae** [comment]
@RolandMacDoland @sungwoo-pasqal shall we turn this example into a tutorial?

**Comment by chMoussa** [comment]
Alright @RolandMacDoland it is converted to a simple tutorial

File: docs/tutorials/qml/classification.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+dataset = IrisDataset()
+
+dataloader = DataLoader(dataset, batch_size=20, shuffle=True)
+
+```
+
+
+## Hybrid QNN
+
+We set up the QNN part composed of multiple feature map layers, each followed by a variational layer.
```
Comment (RolandMacDoland) [inline docs/tutorials/qml/classification.md:61]: This is basically the definition for the HEA right ?

File: docs/tutorials/qml/classification.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+The output will be the expectation value wrt a Z observable on qubit 0
```
Comment (RolandMacDoland) [inline docs/tutorials/qml/classification.md:62]:
```suggestion
The output will be the expectation value with respect to a $Z$ observable on qubit 0
```

File: docs/tutorials/qml/classification.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Then we add a simple linear layer serving as as classification head. Note softmax is not applied with the cross-entropy loss.
```
Comment (RolandMacDoland) [inline docs/tutorials/qml/classification.md:63]: Not sure I totally get that last sentence. Could you elaborate a bit more please ?

Comment (RolandMacDoland) [inline docs/tutorials/qml/classification.md:63]:
```suggestion
Then we add a simple linear layer serving as a classification head. Note softmax is not applied with the cross-entropy loss.
```

File: docs/tutorials/qml/classification.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+## Inference
+
+Finally, we can apply our model on the test set and check the score.
+
+```python exec="on" source="material-block" session="classification"
+X_test, y_test = dataset.X_test, dataset.y_test
+preds_test = torch.argmax(torch.softmax(model(X_test), dim=1), dim=1)
+accuracy_test = (preds_test == y_test).type(torch.float32).mean()
+## Should reach higher than 0.9
+print("Test Accuracy", accuracy_test.item()) # markdown-exec: hide
```
Comment (RolandMacDoland) [inline docs/tutorials/qml/classification.md:117]:
```suggestion
print(f"Test Accuracy: {accuracy_test.item()}") # markdown-exec: hide
```

**Review by RolandMacDoland — COMMENTED** [review]
Thanks @debrevitatevitae. I guess it needs a bit of elaboration, especially for people who aren't so versed in ML (like me).

File: docs/tutorials/qml/classification.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+```
+
+## Training
+
+Then we can set up the training part:
+
+```python exec="on" source="material-block" session="classification"
+opt = torch.optim.Adam(model.parameters(), lr=lr)
+criterion = nn.CrossEntropyLoss()
```
[the remark this answers is not in the document]
Comment (chMoussa) [reply docs/tutorials/qml/classification.md:91]: It is this kind of formula:
- sum_i y_i log(p_i) where you imagine y_i labels in a one-hot encoded form (imagine 3 possible labels, 1 is (0, 0, 1), 2 is (0, 1, 0) and 3 is (1, 0, 0)),  and p_i are probabilities for the correct label given a data instance i. You want to minimize this, which means the p_i should be as closed to a proba 1 for the correct label i.

**Review by RolandMacDoland — APPROVED** [review]
